### PR TITLE
Fnd 1434 teacher join section cap

### PR DESCRIPTION
**Pull request by JamesHankinsDev**
Adds Rails and React functionality to handle section sizes up to 500 followers. Will otherwise reject new students being added and recommend to users to reach out to their teachers for support, or for teacher to break their sections into smaller groups

Errors similar to the below will now become present when trying to add/move/copy students to a section if the addition of those students will put the section over the current section capacity (currently set at 500).

Screen Shot 2021-05-20 at 10 56 57 AM

## Links

- spec: [Problem 1, less Req 4](https://docs.google.com/document/d/1Pl7sOO_TKo9zWH0Qv8_HBKzaIM2JNTXnE5Fq0YYUv3I/edit#heading=h.x757xf8ral1d)
- Jira ticket: [Jira Ticket FND-1529](https://codedotorg.atlassian.net/secure/RapidBoard.jspa?rapidView=28&projectKey=FND&modal=detail&selectedIssue=FND-1434)

## Testing story

Includes new and updated unit tests for adding and transferring students into a section.

## Deployment strategy

N/A

## Follow-up work

N/A

## Privacy

Does push student, teacher, and section ID's to a firehose console, but does not include PII.

## Security

N/A

## Caching

N/A

## PR Checklist:

- [X] Tests provide adequate coverage
- [X] Privacy and Security impacts have been assessed
- [X] Code is well-commented
- [X] New features are translatable or updates will not break translations
- [X] Relevant documentation has been added or updated
- [X] User impact is well-understood and desirable
- [X] Pull Request is labeled appropriately
- [X] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/manageStudents/manageStudentsRedux.js [hunk 1]
```diff
@@ -565,6 +596,30 @@ export default function manageStudents(state = initialState, action) {
...
+        ...state.studentData[action.studentIds[i]],
+        isSaving: false
+      };
+      newState.editingData[action.studentIds[i]] = {
+        ...state.editingData[action.studentIds[i]],
+        isSaving: false
+      };
+    }
+    return newState;
+  }
```
Comment (annaxuphoto) [inline apps/src/templates/manageStudents/manageStudentsRedux.js:622]: For my own understanding, what's this loop doing?

File: apps/test/unit/templates/manageStudents/ManageStudentsTableTest.js [hunk 2]
```diff
@@ -354,5 +357,176 @@ describe('ManageStudentsTable', () => {
...
+                  error: null,
+                  sectionDisplay: '',
+                  verb: 'copy'
+                }}
+              />
+            )
+          ).to.be.true;
+        });
+      });
+    });
```
Comment (annaxuphoto) [inline apps/test/unit/templates/manageStudents/ManageStudentsTableTest.js:644]: nit: can we also add a test to make sure the alert doesn't show when it's not supposed to?

**Review by annaxuphoto — COMMENTED** [review]
Sorry for getting to this late in your time with us! I just have a couple questions, but thanks for doing this work!

File: apps/src/templates/manageStudents/manageStudentsRedux.js (hunk 1 above)
Comment (JamesHankinsDev) [reply apps/src/templates/manageStudents/manageStudentsRedux.js:622]: Sorry for the delay!

This loop I snagged from the `ADD_STUDENT_FAILURE` case. It sets the status of each of the students that failed to add with `{ isSaving: false}`, which allows a user to continue to make edits without refreshing the page. 

Without this loop, an attempt at `save all` disables the individual `save` buttons for a failed bulk add. A user would need to refresh the page and resubmit a bulk add to try again.

With it, the full notification displays, and teachers can add individual students with the `add` buttons, or cancel students to match the spaces available and re-submit a bulk add without refreshing the page.

Hopefully, that makes sense? happy to demo it as well if that's helpful!

File: apps/test/unit/templates/manageStudents/ManageStudentsTableTest.js (hunk 2 above)
Comment (JamesHankinsDev) [reply apps/test/unit/templates/manageStudents/ManageStudentsTableTest.js:644]: I've added these tests as well, but it gets a bit wordy. Happy to refactor/redact some of this if it feels a bit too much here!

**Review by annaxuphoto — APPROVED** [review]
